control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -359,7 +363,14 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 		if err := r.finalizeNonExternalBrokerTopic(broker, securityOption, topicConfig, logger); err != nil {
 			return err
 		}
+
+		// TODO: this should be removed on external topics too...
+		if err := r.removeFinalizerSecret(ctx, finalizerSecret(broker), secret); err != nil {
```

Comment:
For moving outside (needed), we need the secret there, which requires the brokerConfig.

leaking... :sweat_smile: